test/solver_tests/ilp_solver_test.py

```diff
@@ -196,7 +237,7 @@ def test_simple_potential_graph_inference(self):
         )
         max_lca_height = 10
         potential_graph = self.ilp_solver.infer_potential_graph(
-            unique_character_matrix,
+            unique_character_matrix.astype(str),
```

Comment:
Is this casting necessary? Doesn't this function perform this cast already before passing on to the cythonized function?

Reply:
You're right, it's not completely necessary. Thanks!